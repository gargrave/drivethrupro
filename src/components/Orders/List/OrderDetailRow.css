.App .order-item-row .order-item-row-item + .order-item-row-item {
  margin-left: 30px;
  margin-top: 5px;
}

.App table thead tr th,
.App .order-item-row td.order-item-row-item {
  padding: .5rem;
  text-align: center;
}

.App table thead tr th:first-child,
.App .order-item-row td.order-item-row-item:first-child {
  text-align: left;
}

/* override styles of "remove item" button */
.App .order-item-row .remove-item-button {
  /* change coloring to red-ish */
  color: #ab1616;
  border-color: #ab1616;
  /* update display properties so it fits into a <tr> without any weird spacing issues */
  display: inline;
  padding: 9px 14px;
}

.App .order-item-row .remove-item-button:hover {
  background-color: #ab1616;
  color: white;
}

@media only screen and (min-width: 641px) {
  .App .order-item-row td {
    min-width: 160px;
  }
}
