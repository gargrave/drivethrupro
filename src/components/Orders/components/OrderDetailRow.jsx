import React from 'react'
import { number, shape } from 'prop-types'

import { displayAsDollars } from '../../../utils/utils'

const OrderDetailRow = props => {
  const { index, order } = props
  return (
    <div>
      Order #{index} - {order.totalItems} items - {displayAsDollars(order.totalPrice)}
    </div>
  )
}

OrderDetailRow.propTypes = {
  order: shape({
    totalItems: number.isRequired,
    totalPrice: number.isRequired
  }).isRequired,
  index: number.isRequired
}

export default OrderDetailRow
