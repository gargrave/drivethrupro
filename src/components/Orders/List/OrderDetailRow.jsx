import React from 'react'
import { func, number, shape } from 'prop-types'

import { displayAsDollars } from '../../../utils/utils'
import Button from '../../Common/Button'

import './OrderDetailRow.css'

const OrderDetailRow = props => {
  const { index, order } = props
  return (
    <tr className="order-item-row">
      <td className="order-item-row-item">
        {index}
      </td>
      <td className="order-item-row-item">
        {order.totalItems}
      </td>
      <td className="order-item-row-item">
        {displayAsDollars(order.totalPrice)}
      </td>
      <td className="order-item-row-item">
        <Button inline outline success className="complete-order-btn" onClick={props.handleCompleteOrderClick}>
          Complete
        </Button>
        <Button inline outline warning className="edit-order-btn" onClick={props.handleEditOrderClick}>
          Edit
        </Button>
        <Button inline outline danger className="cancel-order-btn" onClick={props.handleCancelOrderClick}>
          Cancel
        </Button>
      </td>
    </tr>
  )
}

OrderDetailRow.propTypes = {
  order: shape({
    totalItems: number.isRequired,
    totalPrice: number.isRequired
  }).isRequired,
  index: number.isRequired,
  handleCompleteOrderClick: func.isRequired,
  handleEditOrderClick: func.isRequired,
  handleCancelOrderClick: func.isRequired
}

export default OrderDetailRow
