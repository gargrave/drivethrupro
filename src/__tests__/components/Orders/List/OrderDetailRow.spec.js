import React from 'react'
import { shallow } from 'enzyme'

import OrderDetailRow from '../../../../components/Orders/List/OrderDetailRow'

describe('OrderDetailRow', () => {
  let props

  beforeEach(() => {
    props = {
      order: {
        totalItems: 0,
        totalPrice: 0
      },
      index: 1,
      handleCompleteOrderClick: jest.fn(),
      handleEditOrderClick: jest.fn(),
      handleCancelOrderClick: jest.fn()
    }
  })

  test('renders correctly', () => {
    const component = shallow(<OrderDetailRow {...props} />)
    expect(component).toMatchSnapshot()
  })

  test('calls the "complete" callback when the button is clicked', () => {
    const component = shallow(<OrderDetailRow {...props} />)
    expect(props.handleCompleteOrderClick.mock.calls.length).toEqual(0)
    component.find('.complete-order-btn').simulate('click')
    expect(props.handleCompleteOrderClick.mock.calls.length).toEqual(1)
  })

  test('calls the "edit" callback when the button is clicked', () => {
    const component = shallow(<OrderDetailRow {...props} />)
    expect(props.handleEditOrderClick.mock.calls.length).toEqual(0)
    component.find('.edit-order-btn').simulate('click')
    expect(props.handleEditOrderClick.mock.calls.length).toEqual(1)
  })

  test('calls the "cancel" callback when the button is clicked', () => {
    const component = shallow(<OrderDetailRow {...props} />)
    expect(props.handleCancelOrderClick.mock.calls.length).toEqual(0)
    component.find('.cancel-order-btn').simulate('click')
    expect(props.handleCancelOrderClick.mock.calls.length).toEqual(1)
  })
})
