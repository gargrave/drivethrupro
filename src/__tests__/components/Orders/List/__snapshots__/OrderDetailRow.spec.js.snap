// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderDetailRow renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <tr
    className="order-item-row"
>
    <td
        className="order-item-row-item"
    >
        1
    </td>
    <td
        className="order-item-row-item"
    />
    <td
        className="order-item-row-item"
    >
        $0.00
    </td>
    <td
        className="order-item-row-item"
    >
        <Button
            className="complete-order-btn"
            inline={true}
            onClick={[Function]}
        >
            Complete
        </Button>
        <Button
            className="edit-order-btn"
            inline={true}
            onClick={[Function]}
            outline={true}
        >
            Edit
        </Button>
        <Button
            className="cancel-order-btn"
            danger={true}
            inline={true}
            onClick={[Function]}
            outline={true}
        >
            Cancel
        </Button>
    </td>
</tr>,
  "nodes": Array [
    <tr
      className="order-item-row"
>
      <td
            className="order-item-row-item"
      >
            1
      </td>
      <td
            className="order-item-row-item"
      />
      <td
            className="order-item-row-item"
      >
            $0.00
      </td>
      <td
            className="order-item-row-item"
      >
            <Button
                  className="complete-order-btn"
                  inline={true}
                  onClick={[Function]}
            >
                  Complete
            </Button>
            <Button
                  className="edit-order-btn"
                  inline={true}
                  onClick={[Function]}
                  outline={true}
            >
                  Edit
            </Button>
            <Button
                  className="cancel-order-btn"
                  danger={true}
                  inline={true}
                  onClick={[Function]}
                  outline={true}
            >
                  Cancel
            </Button>
      </td>
</tr>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <OrderDetailRow
        handleCancelOrderClick={[Function]}
        handleCompleteOrderClick={[Function]}
        handleEditOrderClick={[Function]}
        index={1}
        order={
                Object {
                        "totalItems": 0,
                        "totalPrice": 0,
                      }
        }
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "handleCancelOrderClick": [Function],
          "handleCompleteOrderClick": [Function],
          "handleEditOrderClick": [Function],
          "index": 1,
          "order": Object {
            "totalItems": 0,
            "totalPrice": 0,
          },
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <tr
          className="order-item-row"
>
          <td
                    className="order-item-row-item"
          >
                    1
          </td>
          <td
                    className="order-item-row-item"
          />
          <td
                    className="order-item-row-item"
          >
                    $0.00
          </td>
          <td
                    className="order-item-row-item"
          >
                    <Button
                              className="complete-order-btn"
                              inline={true}
                              onClick={[Function]}
                    >
                              Complete
                    </Button>
                    <Button
                              className="edit-order-btn"
                              inline={true}
                              onClick={[Function]}
                              outline={true}
                    >
                              Edit
                    </Button>
                    <Button
                              className="cancel-order-btn"
                              danger={true}
                              inline={true}
                              onClick={[Function]}
                              outline={true}
                    >
                              Cancel
                    </Button>
          </td>
</tr>,
        "_debugID": 2,
        "_renderedOutput": <tr
          className="order-item-row"
>
          <td
                    className="order-item-row-item"
          >
                    1
          </td>
          <td
                    className="order-item-row-item"
          />
          <td
                    className="order-item-row-item"
          >
                    $0.00
          </td>
          <td
                    className="order-item-row-item"
          >
                    <Button
                              className="complete-order-btn"
                              inline={true}
                              onClick={[Function]}
                    >
                              Complete
                    </Button>
                    <Button
                              className="edit-order-btn"
                              inline={true}
                              onClick={[Function]}
                              outline={true}
                    >
                              Edit
                    </Button>
                    <Button
                              className="cancel-order-btn"
                              danger={true}
                              inline={true}
                              onClick={[Function]}
                              outline={true}
                    >
                              Cancel
                    </Button>
          </td>
</tr>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <OrderDetailRow
    handleCancelOrderClick={[Function]}
    handleCompleteOrderClick={[Function]}
    handleEditOrderClick={[Function]}
    index={1}
    order={
        Object {
            "totalItems": 0,
            "totalPrice": 0,
          }
    }
/>,
}
`;
